### PR TITLE
naming timer sampling and worker threads

### PR DESCRIPTION
**Pull request by ehds**
### What problem does this PR solve?

Issue Number: https://github.com/apache/brpc/issues/1923

Problem Summary: Naming threads for debugging.

线程命名方式： `brpc` 开头 + `线程用途`。
其中 worker 线程以 id 作为区分。

### What is changed and the side effects?
No
Changed:
Naming sampling, timer and worker threads.
Side effects:
- Performance effects(性能影响):
No
- Breaking backward compatibility(向后兼容性): 
No
---
### Check List:
- Please make sure your changes are compilable(请确保你的更改可以通过编译).
- When providing us with a new feature, it is best to add related tests(如果你向我们增加一个新的功能, 请添加相关测试).
- Please follow [Contributor Covenant Code of Conduct](../../master/CODE_OF_CONDUCT.md).(请遵循贡献者准则).

### CODE REVIEW

File: src/bvar/detail/sampler.cpp [hunk 1]
```diff
@@ -108,6 +109,7 @@ class SamplerCollector : public bvar::Reducer<Sampler*, CombineSampler> {
     void run();
 
     static void* sampling_thread(void* arg) {
+        butil::PlatformThread::SetName("brpc_sampling");
```
Comment (wwbmmm) [inline src/bvar/detail/sampler.cpp:112]: In addition to the sampling thread, bvar also has a grab thread and two dump threads. (see variable.cpp and collector.cpp) Can you also add names to them?

Comment (ehds) [reply src/bvar/detail/sampler.cpp]: done

**Comment by chenBright** [comment]
https://github.com/apache/brpc/blob/d6df0f156ad6bd8ec6722a3c3c15691607bb62e7/src/brpc/details/usercode_backup_pool.cpp#L103-L106

Can you also add names to UserCodeBackupPool?

**Comment by ehds** [comment]
> UserCodeBackupPool

done

**Comment by wwbmmm** [comment]
LGTM

**Pull request merged**